p2p/server/server.go

```diff
@@ -362,18 +374,23 @@ func (s *Server) StreamRequest(
 	}
 
 	var srvError *ServerError
-	took := time.Since(start).Seconds()
+	isServerError := errors.As(err, &srvError)
```

Comment:
Why the dedicated variable? It's not uncommon to just have cases with `errors.As` and `errors.Is` to check for specific errors.

Reply:
These were some leftovers, fixed